SMAPP Code Review

christokita opened the pull request.

christokita committed:
Create plot_exposure_diversity.R

christokita committed:
compute average exposure distribution of an article

christokita committed:
show total exposure to fake news by article

christokita committed:
avg belief distribution

christokita committed:
Increase number of posterior samples for follower ideology estimate

christokita committed:
plot exposure and belief data with redone ideol estimate data

christokita committed:
change name of simple ideol distribution estimates

christokita committed:
update path of storage of simple ideol estimates

christokita committed:
Plot follower ideology dist shapes

christokita committed:
Degree distribution plots

christokita committed:
degree distriubtion

christokita committed:
move old ideology of tweeters script

christokita committed:
ideologies of tweeters

christokita committed:
redo average exposure ideology plots

christokita committed:
Update plot_ideology.R

christokita committed:
update belief plots

christokita committed:
update exposure figures

christokita committed:
fix belief and tweet figures

christokita committed:
Update 03a_add_tweet_metadata.py

christokita committed:
add source/target tweet ids to RT network

christokita committed:
Update 04a_construct_retweet_networks.py

christokita committed:
draft script to simulate intervention

christokita committed:
Update theme_ctokita.R

christokita committed:
Update plot_networkmetrics.R

christokita committed:
Update plot_exposure.R

christokita committed:
first run of intervention model

christokita committed:
Create requirements.txt

christokita committed:
add timespread exposure in interventions

christokita committed:
add non-intervention simulation

christokita committed:
Update 08_model_interventions.py

christokita committed:
plot sample intervention data

christokita committed:
re-organize scripts

christokita committed:
Create get_users_to_check_ideologies.py

christokita committed:
include new followers in ideological estimation

christokita committed:
Update 05a_population_ideology_prior.py

christokita committed:
plot interventions with new data

christokita committed:
Update 08_model_interventions.py

christokita committed:
plot RT networks and intervention networks

christokita committed:
model interventions and get sample network

christokita committed:
update population ideology prior method

christokita committed:
reorder scripts

christokita committed:
rename exposure scripts

christokita committed:
udpate exposure script to account for delay in exposure

christokita committed:
Update 06_estimate_belief.py

christokita committed:
update exposure plots

christokita committed:
Update simulate_intervention.cmd

christokita committed:
Update 07a_construct_retweet_networks.py

christokita committed:
Update 04b_follower_ideologies_inference.py

christokita committed:
update intervention data plots

christokita committed:
tweak exposure plots

christokita committed:
plot exposure assumptions

christokita committed:
plot network with exposure

christokita committed:
change plot output from png to pdf

christokita committed:
redo plots in pdf

christokita committed:
fix belief ideol dist plot

christokita committed:
adjust belief plots

christokita committed:
simulate interventions with separate sharing and viz reduction

christokita committed:
Update 08a_model_interventions.py

christokita committed:
replot new intervention simulations

christokita committed:
Update .gitignore

christokita committed:
crowdsourced intervention simulations

christokita committed:
Update 07a_construct_retweet_networks.py

christokita committed:
Update 08a_model_interventions.py

christokita committed:
run and analyze crowd-sourced interventions

christokita committed:
re-do intervention simulations and plots

christokita committed:
update exposure and belief plots

christokita committed:
udpate exposure/belief plots again

christokita committed:
redo belief, exposure, and tweeting plots

christokita committed:
update plots for paper

christokita committed:
redo plots

christokita committed:
intervention and exposure/belief plots

christokita committed:
update figures

christokita committed:
plots

christokita committed:
add belief estimation to intervention code

christokita committed:
plot intervention belief impact

christokita committed:
add direct belief impact for interventions

christokita committed:
update intervention script

christokita committed:
plot intervention belief data

christokita committed:
Update 06_estimate_belief.py

christokita committed:
plot beliefs per exposure over time

christokita committed:
plot total exposure-belief by article/source lean

christokita committed:
plot belief and exposure by 6 hr window of time

christokita committed:
Create data_check_for_kevin.py

christokita committed:
Update data_check_for_kevin.py

christokita committed:
prep expanded-scope-collection data

christokita committed:
add new tweets into tweet_parsing pipeline

christokita committed:
fix handling/prepping of expanded tweet data

christokita committed:
determine which missing tweeters need friend/follower lists

christokita committed:
Update prep_new_data.py

christokita committed:
add friend/followers of missing tweeters

christokita committed:
Update 01_parse_tweets.py

add descriptive comments to files

christokita committed:
Update 01_parse_tweets.py

christokita committed:
Update prep_new_data.py

christokita committed:
Update 02a_parse_users.py

christokita committed:
Update 02a_parse_users.py

christokita committed:
Update 02b_count_users.py

christokita committed:
Create remove_duplicate_friendfollower_lists.py

christokita committed:
reorganize scripts

christokita committed:
Update 03a_clean_ideology_scores.py

christokita committed:
udpate approach to matching article to unmatched tweets

christokita committed:
update slurm scripts with proper activation of conda

christokita committed:
Update 04a_population_ideology_prior.py

christokita committed:
fix ideology sampling script

christokita committed:
update follower ideology inference process

christokita committed:
manually confirm tweets without article ID

christokita committed:
Update 03c_label_unmatched_tweets.py

christokita committed:
udpate compute_exposure.cmd

We now have 149 unique stories represented in our tweets, so we needed to increase the sbatch array to reflect the real number of jobs we need.

christokita committed:
fix up intervention modeling scripts

christokita committed:
Update simulate_intervention.cmd

christokita committed:
redo exposure plots

christokita committed:
do plots

christokita committed:
plot interventions and tweets

christokita committed:
Update requirements.txt

christokita committed:
plot network metrics

christokita committed:
plot networks and ideology

christokita committed:
fix belief reduction in intervention simulations

christokita committed:
Update 08_model_interventions.py

christokita committed:
Update 08_model_interventions.py

christokita committed:
Update 07a_construct_retweet_networks.py

christokita committed:
Update data_check_for_kevin.py

christokita committed:
add one-week window filter to tweet dataset

rename the unmatched tweet labeling script to '03c_finalize_tweet_dataset.py'

christokita committed:
replot tweeting figures

christokita committed:
Update plot_tweeting.R

christokita committed:
plot exposure

christokita committed:
plot belief

christokita committed:
re-plot exposure

christokita committed:
plot exposure-belief

christokita committed:
plot ideology

christokita committed:
plot network metrics

christokita committed:
redo hourbin belief plots

christokita committed:
replot networks

christokita committed:
replot interventions

christokita committed:
remove crowd source interventions plots

christokita committed:
plot interventions

christokita committed:
fix spacing on belief+exposure plot panels

christokita committed:
plot fit of exposure per belief by article lean and source type

christokita committed:
Update plot_interventions.R

christokita committed:
Update plot_belief.R

christokita committed:
plot belief vs exposure proportions

christokita committed:
update belief and exposure figures

christokita committed:
exposure vs belief figs

christokita committed:
check if self-reported and twitter ideologies match up

christokita committed:
Update belief_vs_exposure_articlelean.pdf

christokita committed:
Update 03d_count_users_in_study.py

christokita committed:
fix intervention bar plot axes

christokita committed:
fix cumulative belief figure

christokita committed:
clean up: remove old scripts

christokita committed:
plot cross-ideology exposure

christokita committed:
update name

christokita committed:
Update README.md

christokita committed:
remove Bolstad package from plot_interventions.R

christokita committed:
clean up plot_networkmetrics.R

christokita committed:
Update README.md

christokita committed:
Update README.md

christokita committed:
Update plot_belief.R

christokita committed:
first batch of script header updates

christokita committed:
fix data path names

christokita committed:
fix headers and paths on first batch of scripts

christokita committed:
Update 03a_clean_ideology_scores.py

christokita committed:
new batch of headers fixed

christokita committed:
third batch of new python script headers

christokita committed:
fourth quick batch of script headers

christokita committed:
final batch of main python script headers

christokita committed:
clean up repository and delete old scripts

christokita commented on README.md:
```diff
@@ -1,2 +1,62 @@
+* **Python**: All Python packages for this project can be found in `<requirements.txt>` and can easily be installed with `<pip install -r requirements.txt>` from command line.
+* **R**: dplyr, ggplot2, tidyr, stringr, RColorBrewer, scales, brms
+
+### Data
```
what is a figure session? if you meant figure section, what usually goes into that?

christokita replied:
from Cezar: add figure/output section that explains what each subdirectory within the `output/` folder is.

christokita commented on README.md:
```diff
@@ -1,2 +1,62 @@
+### Structure of the code
+
+#### The main analysis and data visualization scripts
+Python scripts are named with a leading number that indicates the order in which they should be run, e.g., `01_parse_tweets.py`. Thus, the python script are structured in the analysis pipeline for the paper. The general steps of the pipeline are as follows:
```
For clarity: what do you mean exactly? Just trying to make sure I implement correctly.

cezarfpek replied:
It's not necessary, but I meant to add the purpose statement from the files to the read_me as well just to make it clearer

christokita committed:
remove unused tweet_parser.py script

The functionality of this script is already embedded as functions defined within the 01_parse_tweets.py script.

christokita committed:
update main analysis script headers

christokita committed:
add outline of `output/` to README

christokita committed:
remove reference to `scripts/analysis_functions/` from README

christokita committed:
Delete pulling tweet_data.py

christokita committed:
Merge branch 'main' of https://github.com/christokita/news-belief-at-scale

christokita committed:
add descriptors to certain headers about files in/out

christokita committed:
update data paths in scripts

`fake-news-diffusion/` becomes `news-belief-at-scale/`

christokita committed:
fix typo in 01_parse_tweets.py

christokita committed:
fix spelling of "hard drive" in python scripts

christokita commented on scripts/plot_belief.R:
```diff
@@ -0,0 +1,584 @@
+
+
+####################
+# Ideological distributions of belief
```
fixed the bin size. thanks for catching!

christokita commented on scripts/plot_belief.R:
```diff
@@ -0,0 +1,584 @@
+
+
+####################
+# Belief as function of exposure over time
```
Labels* for trend lines are added later. Clarified in script.

christokita commented on scripts/plot_belief.R:
```diff
@@ -0,0 +1,584 @@
+             ncol = 1,
+             strip.position = "top",
+             scales = "free_y") 
+gg_follower_count
```
deleted!

christokita committed:
update plot_belief.R and plots following code review

christokita committed:
update organization of plot_belief.R

christokita committed:
fix header of plot_belief.R

christokita committed:
size -> linewidth for ggplot lines

christokita committed:
fix quotes at head of plot_belief.R

christokita committed:
update plot_exposure.R in response to code review

christokita committed:
Update plot_exposure.R

christokita committed:
fix headers in plot_belief.R and plot_exposure.R

christokita commented on scripts/plot_exposure_diversity.R:
```diff
@@ -0,0 +1,316 @@
+########################################
+#
+# PLOT: Diversity of exposure to news
```
this was an old supplemental analysis. we don't really use it in the paper; however it might be useful to keep it for the review process. 

For now, I've cleaned it up a bit and moved it to the `suppl_code/` subdirectory.

christokita committed:
clean up plot_exposure_diversity.R and move to `suppl_code/`

christokita commented on scripts/plot_exposurebelief_fig.R:
```diff
@@ -0,0 +1,555 @@
+######################################## Total Exposure/Belief ########################################
+
+####################
+# PLOT: Total exposure and belief
```
labels are added in a vector art program later. I'll add a note of that as I did for the regression line labels in other scripts.

christokita commented on scripts/plot_exposurebelief_fig.R:
```diff
@@ -0,0 +1,555 @@
+
+
+####################
+# PLOT: Time to 50 percent exposure/belief
```
Added not explaining this section of code and noting that it is not used in the paper.

christokita committed:
update `plot_exposurebelief_fig.R` according to code review

christokita commented on scripts/plot_ideology.R:
```diff
@@ -0,0 +1,244 @@
+############################## Summary of tweeter ideology ##############################
+
+####################
+# Plot: Total tweets by ideology
```
I've added a comment clarifying why the facet labels are removed.

christokita committed:
udpate plot_ideology.R according to code review

christokita committed:
make graph creation code blocks consistent in format

christokita committed:
plot_ideology.R: fix retweet plot x-axis and y-axis limits on tweeter ideology dist plot

christokita committed:
fix y-axis limits in `plot_ideology.R`

christokita commented on scripts/plot_ideology.R:
```diff
@@ -0,0 +1,244 @@
+
+
+####################
+# Plot: Average ideology distribution of retweets only
```
fixed!

christokita committed:
Update plot_interventions.R

christokita commented on scripts/plot_networkmetrics.R:
```diff
@@ -0,0 +1,307 @@
+  rt_edges %>% 
+  mutate(RT_type = ifelse(grepl("Phantom RT", RT_type), "Direct RT", RT_type),
+         RT_type = gsub(" RT", "", RT_type),
+         RT_type = factor(RT_type, levels = c("Direct", "Indirect", "Self", "Quote"))) %>% 
```
These are all technically four distinct ways of retweeting someone.

users can (a) retweet someone directly, (b) retweet someone indirectly, by retweeting a retweet, (c) retweet themself, or (d) retweet by quote tweeting a tweet. Technically retweeting yourself is a direct retweet of yourself, but we break it out as a different kind of RT. 

I've added a comment clarifying this in the code.

christokita commented on scripts/plot_networkmetrics.R:
```diff
@@ -0,0 +1,307 @@
+############################## Ideology in networks ##############################
+
+####################
+# Plot: Ideological diversity by article veracity
```
thank you!

christokita committed:
udpate plot_networkmetrics.R according to code review

christokita committed:
fix up `plot_tweet.R` according to code review

christokita committed:
tying up loose ends on plotting scripts

christokita committed:
re-run source_type plots in `plot_networkmetrics.R`

christokita committed:
double check avg ideological distribution plots

christokita committed:
Update 03d_count_users_in_study.py

christokita committed:
update figures for average and total ideological exposure/belief

christokita committed:
fix denominator on avg users exposed/believing figures

christokita committed:
add combined belief/exposure figure broken out by source type

christokita committed:
re-do belief plots with correct labels for source type

christokita committed:
update figures: "belief" -> "receptivity"

christokita committed:
update belief-per-exposure figure y axis label

christokita committed:
Update README.md

christokita committed:
Merge pull request #2 from christokita/christokita-patch-1

Update README.md

christokita committed:
use skew normal distribution in follower ideology inference

change our inference of a given user's follower ideologies from strictly gaussian to a skew normal distribution.

christokita committed:
estimate exposed ideologies with skew normal

draw missing ideologies from skew normal (instead of normal) distribution using our new method of inferring follower ideologies as skew normal distributions.

christokita committed:
fix specification of skewnorm in scipy.stats

christokita committed:
clean up belief_file path in 06_estimate_belief.py

christokita committed:
Update 06_estimate_belief.py

christokita committed:
rerun expsoure belief plots

christokita committed:
Update 08_model_interventions.py

christokita committed:
redo belief plots

christokita committed:
redo exposure plots

christokita committed:
redo tweeter ideology plots

christokita committed:
redo network metrics plots

christokita committed:
redo tweeting plots

christokita committed:
redo exposure plots

christokita committed:
Update plot_exposure.R

christokita committed:
redo exposure diversity plots

christokita committed:
fix sd calculation for follower shape plot

christokita committed:
Update 08_model_interventions.py

christokita committed:
Update 08_model_interventions.py

christokita committed:
update intervention plots

christokita committed:
replot interventions

christokita committed:
replot self-reported vs twitter ideology

christokita committed:
replot self-reported vs twitter ideology using synthetic groups

christokita committed:
ideological extremity plots

christokita committed:
fix plots for exposure vs receptivity

christokita committed:
fix tweeter ideology plots

christokita committed:
replot ideological extremity

christokita committed:
Merge pull request #3 from christokita/ideology_estimation_skewed

Add ideological estimation of follower ideologies

christokita committed:
Plot normalized exposure and belief

christokita committed:
plot ideology confusion matrix

christokita committed:
calculate extra stats for belief-per-exposure over time

christokita committed:
Update plot_belief.R

christokita committed:
finish updating exposure per belief plots

christokita committed:
Add figures to compare against baseline exposure

christokita committed:
new figure of receptivity measure over baseline exposure

christokita committed:
Update README.md

christokita committed:
Merge pull request #4 from christokita/christokita-patch-1

Update README.md

christokita committed:
Update README.md

christokita committed:
Update README.md

christokita committed:
Update README.md

christokita committed:
clean up paths in scripts/03a_clean_ideology_scores.py

christokita committed:
Merge branch 'main' of https://github.com/christokita/news-belief-at-scale

christokita committed:
Update README.md

christokita committed:
Update README.md

christokita committed:
Update README.md

christokita committed:
Update README.md

christokita committed:
Update README.md

christokita committed:
Update README.md
